src/components/autocompleter/autocompleter.jsx

```diff
@@ -0,0 +1,122 @@
+import PropTypes from 'prop-types';
+import React, { forwardRef, useEffect, useState } from 'react';
+import useFetch from '@bloodyaugust/use-fetch';
+import useMounted from '../../hooks/use-mounted.js';
+import Select from '../select/select.jsx';
+import ListOption from '../list-option/list-option.jsx';
+import mockConstants from '../../mocks/mock-constants.js';
+
+const { API_ROOT } = mockConstants;
+
+const Autocompleter = forwardRef(function Autocompleter(props, ref) {
+  const { execute } = useFetch();
+  const mounted = useMounted();
+  const [models, setModels] = useState(props.models);
+
+  useEffect(() => {
+    if (!mounted.current) return;
+
+    setModels(props.models);
+  }, [props.models]);
+
+  useEffect(fetchModels, []);
+
+  function fetchModels(searchString) {
+    if (!props.apiEndpoint) { return; }
+
+    execute(apiEndpoint(searchString))
+      .then((respObj) => {
+        return setModels(respObj.json.results.map(result => respObj.json[result.key][result.id]));
+      }).catch((error) => {
+        if (error.error && error.error.type !== 'aborted') {
+          throw error;
+        }
+      });
+  }
+
+  function apiEndpoint(searchString) {
+    if (searchString) {
+      return `${API_ROOT}${props.apiEndpoint}?${props.searchParam}=${searchString}`;
+    }
+
+    return `${API_ROOT}${props.apiEndpoint}`;
+  }
+
+  function onInput(event) {
+    fetchModels(event.target.value);
+  }
+
+  const listOptionRefs = models.map(() => React.createRef());
+  const listOptionElements = models.map((modelInfo, index) => {
+    return (
+      <ListOption key={modelInfo.id} ref={listOptionRefs[index]} value={modelInfo}>
+        {props.displayValueEvaluator(modelInfo)}
+      </ListOption>
+    );
+  });
+
+  return (
+    <Select
+      label={props.label}
+      id={props.id}
+      name={props.name}
+      className={props.className}
+      listOptionRefs={listOptionRefs}
+      displayValueEvaluator={props.displayValueEvaluator}
+      onChange={props.onChange}
+      onInput={onInput}
+      required={props.required}
+      readOnly={props.readOnly}
+      placeholder={props.placeholder}
+      errorText={props.validationMessage}
+      value={props.value}
+      ref={ref}
+    >
+      {listOptionElements}
+    </Select>
+  );
+});
+
+function displayName(modelInfo) {
+  return modelInfo.title || modelInfo.name || modelInfo.full_name || modelInfo.currency;
+}
+
+Autocompleter.propTypes = {
+  // `apiEndpoint` should be the route of the api's endpoint (excluding the base api), eg. `/workspaces`.
```

Comment:
If we want this to be on the documentation page, you need to use styleguidist comments:

```suggestion
  /** `apiEndpoint` should be the route of the api's endpoint (excluding the base api), eg. `/workspaces`. */
```

Same for the rest of these comments in the prop-types interface.